### PR TITLE
refactor: use beanContext instead of applicationcontext

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

**Review by musketyr — APPROVED** [review]
I really like this. You are right. One should use the minimal interface required.

Is the Inject annotation used?

**Comment by sdelamo** [comment]
> Is the Inject annotation used?

Not sure I understand. It is using constructor injection. This is a breaking change. You could leave the old constructor and annotate the new one with `@Inject`

File: libs/micronaut-snitch/src/main/java/com/agorapulse/micronaut/snitch/SnitchInterceptor.java [hunk 1]
```diff
@@ -20,8 +20,10 @@
 import io.micronaut.aop.MethodInterceptor;
 import io.micronaut.aop.MethodInvocationContext;
 import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.BeanContext;
 import io.micronaut.inject.qualifiers.Qualifiers;
 
+import javax.inject.Inject;
```
Comment (musketyr) [inline libs/micronaut-snitch/src/main/java/com/agorapulse/micronaut/snitch/SnitchInterceptor.java]: @sdelamo here there is a new import which looks unused

Comment (sdelamo) [reply libs/micronaut-snitch/src/main/java/com/agorapulse/micronaut/snitch/SnitchInterceptor.java]: That it is a mistake.

**Pull request merged**